kbase-extension/static/kbase/js/common/cellComponents/paramsWidget.js

```diff
@@ -387,103 +390,114 @@ define([
                 return param.id;
             });
 
-            const layout = orderedParams.map(function (parameterId) {
-                const elementId = html.genId();
-                const advanced = paramMap[parameterId].ui.advanced ? parameterId : false;
+            const layout = orderedParams
+                .map(function (parameterId) {
+                    const elementId = html.genId();
+                    const advanced = paramMap[parameterId].ui.advanced ? parameterId : false;
 
-                view[parameterId] = {
-                    id: elementId
-                };
+                    view[parameterId] = {
+                        id: elementId,
+                    };
 
-                return div({
-                    id: elementId,
-                    dataParameter: parameterId,
-                    dataAdvancedParameter: advanced
-                });
-            }).join('\n');
+                    return div({
+                        id: elementId,
+                        dataParameter: parameterId,
+                        dataAdvancedParameter: advanced,
+                    });
+                })
+                .join('\n');
 
             return {
                 content: layout,
                 layout: orderedParams,
                 params: params,
                 view: view,
-                paramMap: paramMap
+                paramMap: paramMap,
             };
         }
 
-        // LIFECYCLE API
-        function renderParameters() {
-            // First get the app specs, which is stashed in the model,
-            // with the parameters returned.
-            // Separate out the params into the primary groups.
-            const appSpec = model.getItem('appSpec');
-
-            return Promise.try(function () {
-                const params = model.getItem('parameters');
-                let parameterParams = makeParamsLayout(
-                    params.layout.filter(function (id) {
-                        const original = params.specs[id].original;
-
-                        let isParameter = false;
+        function createParameterWidget(appSpec, parameterInfo, parameterId) {
```

Comment:
here's where my changes start. i broke this out of the renderParameters() function to have it be it's own thing. I also changed it to remove an unnecessary Promise block, since makeFieldWidget returns a promise already all we have to do here is catch the error